src/expr/src/scalar/func.rs

```diff
@@ -1457,9 +1518,34 @@ fn exp_dec<'a>(a: Datum<'a>, scale: u8) -> Result<Datum<'a>, EvalError> {
     Ok(cast_float64_to_decimal(Datum::from(a.exp()), scale)?)
 }
 
+fn exp_apd<'a>(a: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let mut a = a.unwrap_apd();
+    let mut cx = apd::cx_datum();
+    cx.exp(&mut a.0);
+    let cx_status = cx.status();
+    if cx_status.overflow() {
+        Err(EvalError::FloatOverflow)
+    } else if cx_status.subnormal() {
+        Err(EvalError::FloatUnderflow)
+    } else {
+        apd::munge_apd(&mut a.0).unwrap();
+        Ok(Datum::APD(a))
+    }
+}
+
 fn power<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
```

Comment:
```POWER (0,0)``` appears to be problematic (https://en.wikipedia.org/wiki/Zero_to_the_power_of_zero#Treatment_on_computers), but the consensus between Postgres, MySQL, SQL Server and Oracle is that the answer should be ```1``` . We return```NaN```:

```
materialize=> SELECT POW ( '0'::APD, '0'::apd);
 pow 
-----
 NaN
(1 row)
```

Reply:
Oh, great catch--tysm. Fixed this.